### ISSUE TITLE
Screen Reader: Sign in: The updated error message is not announced during typing

### ISSUE THREAD

**Issue by elisenwazuoke** [Expensify/App#77257]
## Action Performed:
Prerequisites:
- the user is not signed in

1. Using Android TalkBack or iOS VoiceOver, open the Expensify app
2. Swipe to the 'Continue' button and double-tap to activate
3. Note that the error message appeared on the screen
4. Swipe to the 'Phone or email' input field and start to type.
5. Note that the error message is updated
6. Note what is announced.

## Expected Result:
The updated error message should be announced.

## Actual Result:
The updated error message is not announced during typing.

## Area issue was found in:
Sign in

## Failed WCAG checkpoints
WCAG 4.1.3 Status Messages
https://www.w3.org/WAI/WCAG22/Understanding/status-messages

## User impact:
When appropriate roles or properties are assigned to status messages, the new content is spoken by screen readers in such a way as to assist blind and low vision users. Assigning proper roles or properties to status messages provides possible future uses and personalization opportunities, such as the potential to be exploited by assistive technologies created for users with some cognitive disabilities.

## Suggested resolution:
Ensure the updated error message is announced once it is changed visually.
https://appt.org/en/docs/react-native/samples/accessibility-live-region

## Environment(s)
Samsung Galaxy S24 Ultra - Android 14, Partner - Native (No mobile browser)

## Screenshots/Videos
![Image](https://github.com/user-attachments/assets/c007c9aa-8e80-4dd5-a052-025c78bbaec1)

https://github.com/user-attachments/assets/e8bb604a-a12c-49fd-80e3-9ac2c158a4e9

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7005380
**Issue reported by:** NA
**Slack conversation:** NA

<details><summary>Issue Owner</summary>Current Issue Owner: @arosiclair</details>
